### PR TITLE
Mtb example - Not ready to merge

### PR DESCRIPTION
**Pull request by nategbrown9**

### CODE REVIEW

File: assessmentModel/src/commonMain/kotlin/northwestern/MtbAssessment.kt [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        FlankerInstructionStep::class with FlankerInstructionStep.serializer()
+        FlankerForm::class with FlankerForm.serializer()
+        //FlankerParentForm::class with FlankerParentForm.serializer()
+        FlankerInstructionForm::class with FlankerInstructionForm.serializer()
+        IntIntChoiceInputField::class with IntIntChoiceInputField.serializer()
+    }
+    polymorphic(Assessment::class) {
+        MtbAssessmentObject::class with MtbAssessmentObject.serializer()
+    }
+    polymorphic(Question::class) {
```
Comment (syoung-smallwisdom) [inline assessmentModel/src/commonMain/kotlin/northwestern/MtbAssessment.kt:40]: Nit: Do you need this? It's really only there in our code for documentation.

File: assessmentModel/src/commonMain/kotlin/northwestern/MtbAssessment.kt [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    @SerialName("steps")
+    override val children: List<Node>,
+    override val versionString: String? = null,
+    override val resultIdentifier: String? = null,
+    override var estimatedMinutes: Int = 0,
+    val taskOrientation: String = "portrait",
+    val defaultSteps: MtbDefaultSteps? = null,
+    val stepRules: List<MtbStepRule> = listOf()
+
+
```
Comment (syoung-smallwisdom) [inline assessmentModel/src/commonMain/kotlin/northwestern/MtbAssessment.kt:74]: Nit: Empty spaces. When Kotlin does the formatting, the spaces aren't here.

File: assessmentModel/src/commonMain/kotlin/northwestern/MtbAssessment.kt [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    //Could remove this and use above "flankerParentForm" instead
+    val steps: List<Node> = listOf()
+) : MtbStep(), FormStep
+
+
+@Serializable
+@SerialName("flankerInstructionForm")
+data class FlankerInstructionForm(
+    override val identifier: String,
+    val isInstruction: Boolean = true,
```
Comment (syoung-smallwisdom) [inline assessmentModel/src/commonMain/kotlin/northwestern/MtbAssessment.kt:117]: Nit: Things like `isInstruction` and `isPractice` are very "Swift-like" syntax. Is there a better word choice on Kotlin?

For example, I moved away from `isExclusive` and `isOptional` to `exclusive` and `optional` to match the Json coming from Alx's POJO objects.

File: assessmentModel/src/commonMain/kotlin/northwestern/MtbAssessment.kt [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+    override val resultIdentifier: String?
+        get() = null
+    override val comment: String?
+        get() = null
+    override val hideButtons: List<ButtonAction>
+        get() = listOf()
+    override val buttonMap: Map<ButtonAction, ButtonActionInfo>
+        get() = mapOf()
+
+}
```
Comment (syoung-smallwisdom) [inline assessmentModel/src/commonMain/kotlin/northwestern/MtbAssessment.kt:156]: Nit: Space before

File: assessmentModel/src/commonMain/kotlin/northwestern/MtbAssessment.kt [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+) : MtbStep(), FormStep
+
+
+@Serializable
+@SerialName("flankerInstructionForm")
+data class FlankerInstructionForm(
+    override val identifier: String,
+    val isInstruction: Boolean = true,
+    @SerialName("inputFields")
+    override val children: List<Node> = listOf(),
```
Comment (syoung-smallwisdom) [inline assessmentModel/src/commonMain/kotlin/northwestern/MtbAssessment.kt:119]: Spaces here but not below. Bit inconsistent.

**Comment by nategbrown9** [comment]
> Looks good. Some nits around consistent formatting and spacing.
> 
> Also, would be nice to see some unit tests for the serialization. I find that an easier way to grok the Json <--> POKO translation.
> 
> And, would want to know why build is failing.

The build is failing when it is trying to build for iOS. There is a bug that prevents extending of abstract classes from different files.  Found 2 issues tracking this:
https://github.com/Kotlin/kotlinx.serialization/issues/768
https://github.com/JetBrains/kotlin-native/issues/3900

Seems to work fine for Android, so long as they are in the same module.

**Pull request closed without merging**